### PR TITLE
Added subcategory mappings

### PR DESCRIPTION
**Pull request by vsoni1011**

### CODE REVIEW

File: src/main/java/com/ekart/model/SubCategory.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @ManyToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "category_id")
+    @JsonIgnore
+    private Category category;
+
+    @ManyToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "size_id")
+    @JsonIgnore
+    private Size size;
```
Comment (prap12) [inline src/main/java/com/ekart/model/SubCategory.java:45]: i think this should be many to many or one to many relation. many sizes can be part of one sub category

File: src/main/java/com/ekart/controller/SubCategoryController.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @GetMapping(path = "/{id}")
+    public ResponseEntity<SubCategory> getSubCategoryById(@PathVariable Long id) throws RecordNotFoundException {
+        SubCategory subCategory = subCategoryService.getSubCategoryById(id);
+        return new ResponseEntity<SubCategory>(subCategory, new HttpHeaders(), HttpStatus.OK);
+    }
+
+    @PostMapping
+    public ResponseEntity<SubCategory> addSubCategory(@RequestBody SubCategory subCategory) {
```
Comment (prap12) [inline src/main/java/com/ekart/controller/SubCategoryController.java:43]: add/delete functionality may not be required, as we wont have any corresponding functionality in UI. maybe just as a template we can keep this for now

File: src/main/java/com/ekart/model/SubCategory.java (hunk 1 above)
Comment (vsoni1011) [reply src/main/java/com/ekart/model/SubCategory.java:45]: addressed

File: src/main/java/com/ekart/controller/CategoryController.java [hunk 3]
```diff
@@ -39,15 +39,15 @@
...
-        Category newCategory = categoryService.saveCategory(category);
-        return new ResponseEntity<Category>(newCategory, new HttpHeaders(), HttpStatus.CREATED);
-    }
-
-    @DeleteMapping(path = "/{id}")
-    public HttpStatus deleteCategoryById(@PathVariable Long id) throws RecordNotFoundException {
-        categoryService.deleteCategoryById(id);
-        return HttpStatus.FORBIDDEN;
-    }
+//    @PostMapping
```
Comment (prap12) [inline src/main/java/com/ekart/controller/CategoryController.java:42]: just remove it instead of comment. and also remove from service layer

**Review by prap12 — APPROVED** [review]

**Pull request merged**